chainlibpy/cro_coin.py

```diff
@@ -0,0 +1,174 @@
+import decimal
+from typing import Union
+
+from chainlibpy.generated.cosmos.base.v1beta1.coin_pb2 import Coin
+from chainlibpy.grpc_client import NetworkConfig
+
+from .utils import is_integer
+
+MAX_CRO_SUPPLY = 30_000_000_000  # max CRO supply: 30 billion
+
+
+class CROCoin:
+    def __init__(
+        self,
+        amount: "Union[int, float, str, decimal.Decimal]",
+        unit: str,
+        network_config: "NetworkConfig",
```

Comment:
Sorry. I don't understand this comment.

If you are talking about `"NetworkConfig"` this part, surrounding with `"` is for the quick suggestion purpose in code editor without introducing the circular importing issue.

Reply:
I mean remove unit, and make `self.unit = network_config.coin_base_demon`

Reply:
When I design this class, I am thinking of asking the user to specify the amount together with the unit. 

For example:
```python
# 10 cro should be represented as following
coin = CroCoin("10", "cro", CRO_NETWORK["mainnet"])

# 10000 basecro should be represented as following
coin = CroCoin("10000", "basecro", CRO_NETWORK["mainnet"])
```

Because I feel the `base_denom` has many 0s in front which is not very human friendly, but the `original_denom` is not code friendly. Hence, I specifically ask the user to enter the unit together with the amount, which forms the coin amount.

With this flexibility to user input, if a user would like to represent 10 cro, the user does not need to do a conversion beforehand like below:
```python
amount = 10 * 10 ** 8
coin = CroCoin(amount, "basecro", CRO_NETWORK["mainnet"])
```
Another reason is that Python expresses large or small decimal numbers in the scientific notation, for example, if I would like to port this class for cronos usage in the future, `0.5 * 10 ** 18` will be represented as `5e+17`, which causes some troubles when using this class.